src/main/java/core/basesyntax/JavaDateTimeApi.java

```diff
@@ -1,107 +1,153 @@
 package core.basesyntax;
 
+import java.time.DateTimeException;
 import java.time.LocalDate;
 import java.time.LocalDateTime;
 import java.time.LocalTime;
+import java.time.Month;
 import java.time.OffsetDateTime;
+import java.time.ZoneId;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Locale;
 import java.util.Optional;
 
 public class JavaDateTimeApi {
+    private static final String OFFSET_UKR = "+02:00";
+
     /**
      * Return the current date as a String depending on a query.
-     *
+     * <p>
      * The query can be passed for the whole date or for it's part:
-     *      - FULL - current date as a whole: year, month, day of month
-     *        formatted as `YYYY-MM-DD` (a default return value);
-     *      - YEAR - current year;
-     *      - MONTH - name of the current month;
-     *      - DAY - current day of month;
+     * - FULL - current date as a whole: year, month, day of month
+     * formatted as `YYYY-MM-DD` (a default return value);
+     * - YEAR - current year;
+     * - MONTH - name of the current month;
+     * - DAY - current day of month;
      * In any other case throw DateTimeException.
      **/
     public String todayDate(DateTimePart datePart) {
-        return "Today";
+        LocalDate currentDate = LocalDate.now();
+        switch (datePart) {
+            case FULL:
+                return currentDate.toString();
+            case YEAR:
+                return String.valueOf(currentDate.getYear());
+            case MONTH:
+                return String.valueOf(currentDate.getMonth());
+            case DAY:
+                return String.valueOf(currentDate.getDayOfMonth());
+            default:
+                throw new DateTimeException("Not correct input " + datePart);
+        }
     }
 
     /**
      * Given an Array of 3 elements, where
-     *         - 1-st element is a `year`;
-     *         - 2-nd element is s `month`;
-     *         - 3-rd element is a `day of month`;
-     *
+     * - 1-st element is a `year`;
+     * - 2-nd element is s `month`;
+     * - 3-rd element is a `day of month`;
+     * <p>
      * Return Optional of a date built from these elements.
      */
     public Optional<LocalDate> getDate(Integer[] dateParams) {
-        return Optional.empty();
+        if (dateParams.length < 3) {
+            return Optional.empty();
+        }
+        try {
+            if (dateParams[2] > 31
+                    && Integer.parseInt(String.valueOf(Month
+                    .valueOf(String.valueOf(dateParams[1])))) > 12) {
+                return Optional.empty();
```

Comment:
if you'll get some negative value in the array, let's catch DateTimeParseException